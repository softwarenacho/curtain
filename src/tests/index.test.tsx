import React from 'react';
import renderer from 'react-test-renderer';
import { describe, expect, it } from 'vitest';
import Curtain from '../components/index';

describe('Greeting component', () => {
  const control = false;
  const setControl = () => {};
  it('Greeting component renders correctly', () => {
    const component = renderer.create(
      <Curtain control={control} setControl={setControl} props={{}}>
        <h1>Content</h1>
      </Curtain>,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(' prop working', () => {
    const component = renderer.create(
      <Curtain
        control={control}
        setControl={setControl}
        props={{ backgroundColor: 'white' }}
      >
        <h1>Content</h1>
      </Curtain>,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
