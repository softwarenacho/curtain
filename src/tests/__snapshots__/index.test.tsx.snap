// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Greeting component >  prop working 1`] = `
<div
  className="container"
>
  <div
    className="top"
    onClick={[Function]}
    style={
      {
        "background": "white",
        "borderBottomWidth": "10px",
        "borderColor": "#daa520",
        "borderStyle": "double",
        "borderTopWidth": "",
        "bottom": "unset",
        "filter": "drop-shadow(0px 1px 1px black)",
        "top": "0",
      }
    }
  />
  <div
    className="content"
  >
    <h1>
      Content
    </h1>
  </div>
  <div
    className="bottom"
    onClick={[Function]}
    style={
      {
        "background": "white",
        "borderBottomWidth": "",
        "borderColor": "#daa520",
        "borderStyle": "double",
        "borderTopWidth": "10px",
        "bottom": "0",
        "filter": "drop-shadow(0px -1px 1px black)",
        "top": "unset",
      }
    }
  />
</div>
`;

exports[`Greeting component > Greeting component renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="top"
    onClick={[Function]}
    style={
      {
        "background": "#8b4513",
        "borderBottomWidth": "10px",
        "borderColor": "#daa520",
        "borderStyle": "double",
        "borderTopWidth": "",
        "bottom": "unset",
        "filter": "drop-shadow(0px 1px 1px black)",
        "top": "0",
      }
    }
  />
  <div
    className="content"
  >
    <h1>
      Content
    </h1>
  </div>
  <div
    className="bottom"
    onClick={[Function]}
    style={
      {
        "background": "#8b4513",
        "borderBottomWidth": "",
        "borderColor": "#daa520",
        "borderStyle": "double",
        "borderTopWidth": "10px",
        "bottom": "0",
        "filter": "drop-shadow(0px -1px 1px black)",
        "top": "unset",
      }
    }
  />
</div>
`;
